src/sage/schemes/curves/curve.py

```diff
@@ -246,12 +238,22 @@ def geometric_genus(self):
             sage: C.geometric_genus()
             3
 
+        Note that the geometric genus is only defined for `geometrically
+        irreducible curve <https://stacks.math.columbia.edu/tag/0BYE>`_. This
+        method does not check the condition. Be warned that you may get a
+        nonsensical result if the curve is not geometrically irreducible.
```

Comment:
Perhaps better as a `.. NOTE::` or `.. WARNING::`??

Reply:
OK. I put it to `.. WARNING::`.